src/libs/antares/writer/zip_writer.cpp

```diff
@@ -85,32 +84,29 @@ void ZipWriteJob<ContentT>::writeEntry()
 
     auto file_info = createInfo(pEntryPath);
 
-    Benchmarking::Timer timer_wait;
-    std::lock_guard guard(pZipMutex); // Wait
-    timer_wait.stop();
-    pDurationCollector.addDuration("zip_wait", timer_wait.get_duration());
-
-    Benchmarking::Timer timer_write;
-
-    if (int32_t ret = mz_zip_writer_entry_open(pZipHandle, file_info.get()); ret != MZ_OK)
-    {
-        logErrorAndThrow("Error opening entry " + pEntryPath + " (" + std::to_string(ret) + ")");
-    }
-    int32_t bw = mz_zip_writer_entry_write(pZipHandle, pContent.data(), pContent.size());
-    if (static_cast<unsigned int>(bw) != pContent.size())
-    {
-        logErrorAndThrow("Error writing entry " + pEntryPath + "(written = " + std::to_string(bw)
-                                                + ", size = " + std::to_string(pContent.size()) + ")");
-    }
-
-    timer_write.stop();
-    pDurationCollector.addDuration("zip_write", timer_write.get_duration());
+    pDurationCollector("zip_wait") << [&] {
+        pZipMutex.lock(); // Wait
+    };
+
+    pDurationCollector("zip_write") << [&] {
+        if (int32_t ret = mz_zip_writer_entry_open(pZipHandle, file_info.get()); ret != MZ_OK)
+        {
+            logErrorAndThrow("Error opening entry " + pEntryPath + " (" + std::to_string(ret) + ")");
+        }
+        int32_t bw = mz_zip_writer_entry_write(pZipHandle, pContent.data(), pContent.size());
+        if (static_cast<unsigned int>(bw) != pContent.size())
+        {
+            logErrorAndThrow("Error writing entry " + pEntryPath + "(written = " + std::to_string(bw)
+                                                    + ", size = " + std::to_string(pContent.size()) + ")");
+        }
+    };
+    pZipMutex.unlock();
```

Comment:
Here we need to handle the case where an exception is thrown: we need to catch exceptions, release the mutex, and re-throw.

Or let's not use the new syntax here, it may be simper to keep the old way.